tools/rimage/src/manifest.c

```diff
@@ -225,6 +225,7 @@ static int man_get_module_manifest(struct image *image, struct manifest_module *
 	memcpy(man_module->name, sof_mod->module.name, SOF_MAN_MOD_NAME_LEN);
 	memcpy(man_module->uuid, sof_mod->module.uuid, 16);
 	man_module->affinity_mask = sof_mod->module.affinity_mask;
+	man_module->instance_max_count = sof_mod->module.instance_max_count;
```

Comment:
I don't understand why we overwrite here the information from a toml files

Reply:
@softwarecki sorry, I don't see any overwriting here? AFAIU `man_module` is being created here, it doesn't contain any information yet.

Reply:
Module manifest is read from a toml file. Next it is copied to module manifest structure in the `man_create_modules` function.
https://github.com/thesofproject/sof/blob/b58df6d7607c58bbc91c9b746f847523c38c57e0/tools/rimage/src/manifest.c#L592-L594
then `man_module_create` or `man_module_create_reloc` function is called, which finally call `man_get_module_manifest`. So you overwrite values configured in a toml file.

Reply:
@softwarecki ok, why are other fields here overwritten - even the name and the UUID?

Reply:
@lyakh I don't know. We have the same information in the structure and in the `toml` file. I was hoping that someone who knows why it is done this way will comment.